web_widget_json_graph/static/src/js/web_widget_json_graph.js

```diff
@@ -1,41 +1,23 @@
 odoo.define("web.web_widget_json_graph", function (require) {
     "use strict";
 
+    var ajax = require("web.ajax");
     var AbstractField = require("web.AbstractField");
     var field_registry = require("web.field_registry");
 
     var JSONGraphWidget = AbstractField.extend({
-        template: "JSONGraph",
-        cssLibs: ["/web/static/lib/nvd3/nv.d3.css"],
-        jsLibs: [
-            "/web/static/lib/nvd3/d3.v3.js",
-            "/web/static/lib/nvd3/nv.d3.js",
-            "/web/static/src/js/libs/nvd3.js",
-        ],
-        _render: function () {
+        jsLibs: ["/web/static/lib/Chart/Chart.js"],
+        willStart: function () {
+            this._super();
+            return ajax.loadLibs(this);
+        },
+        start: function () {
             var info = JSON.parse(this.value);
-            /* Jsl:ignore*/
-            /*
-            Ignoring lint erros caused by nv and d3 variables from NVD3.js
-            */
-            if (info) {
-                nv.addGraph(function () {
-                    var chart = nv.models.lineChart().useInteractiveGuideline(true);
-                    chart.xAxis.axisLabel(info.label_x).tickFormat(d3.format(",r"));
-
-                    chart.yAxis.axisLabel(info.label_y).tickFormat(d3.format(".02f"));
-
-                    d3.select(".nv_content svg")
-                        .datum(info.data)
-                        .transition()
-                        .duration(500)
-                        .call(chart);
-                    nv.utils.windowResize(chart.update);
-
-                    return chart;
-                });
-            }
-            /* Jsl:end*/
+            var cx = document.createElement("canvas");
```

Comment:
Just a question, here you are creating another element `canvas` but you already have it in the template `JSONGraph` why don't you use it like thi
ese examples:

[web_kanban_gauge](https://github.com/odoo/odoo/blob/abd9d7a74c91f4632550a376fe566ac28e64377b/addons/web_kanban_gauge/static/src/js/kanban_gauge.js#L117)

[website_links](https://github.com/odoo/odoo/blob/abd9d7a74c91f4632550a376fe566ac28e64377b/addons/website_links/static/src/js/website_links_charts.js#L65)

Reply:
Good suggestion